mipengine/udfgen/udfgenerator.py

```diff
@@ -771,6 +771,38 @@ def udf_logger():
     return UDFLoggerType()
 
 
+class PlaceholrderType(InputType):
+    def __init__(self, name):
+        self.name = "$" + name
+
+
+def placeholder(name):
```

Comment:
Maybe this should be a private method because there is no point of someone trying to use it without editing the NODE service as well.

Reply:
You are right this should be private, and it is because it's not imported in `mipengine/udfgen/__init__.py`. 

There is a huge number of private functions/classes in `mipengine/udfgen/udfgenerator.py` and I don't want to clutter the names by prepending `_`. Instead, all public objects are imported in `mipengine/udfgen/__init__.py` and users should import with `from mipengine.udfgen import ...`. 

This is, I think, a good practice. The internal structure of the subpackage might change. What is in its `__init__.py` is the package interface and should be more stable. Also, if at some point I want to make something public/private, I just need to include/exclude it in `__init__.py`, instead of changing its name everywhere it is used. 

This is how it's done in many well known python libraries. For example in pandas you do `from pandas import DataFrame`, not `from pandas.core.frame import DataFrame`.

In our case, I see that the majority of algorithms do import from `mipengine.udfgen` but not all, and that's wrong and should be fixed.